fix: change input values

KimJaea opened the pull request:
I changed input values from arrays of vector to arrays of arrays of vectors.

KimJaea committed:
fix: delete console log

KimJaea committed:
fix: change input values

originally, this node calculated cosine similarity between two vectors.
now this node calculate cosine similarity between two arrays of vectors.
so this node return arrays of cosine similarity.

Gitgloo commented on GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js:
```diff
@@ -2,30 +2,43 @@ module.exports = function (RED) {
+        magnitudeB = Math.sqrt(magnitudeB);
+
+        if (magnitudeA === 0 || magnitudeB === 0) {
+          throw new Error("Can't calculate cosine similarity of 0 Vector.");
```
If an error occurs, node-red terminates. 😥

leecr1215 replied:
because of throwing err

KimJaea replied:
Thanks, I fixed it. 👍 Please check recent commit below.

chickennight reviewed (dismissed):
well done change, LGTM 👍

hy06ix commented on GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js:
```diff
@@ -40,25 +53,23 @@ module.exports = function (RED) {
-      console.log(Object.keys(stored_vector).length);
       var result = 1;
-      if (typeof input_vector == "undefined" || input_vector == null || input_vector == "") {
+      if (typeof input_vectors == "undefined" || input_vectors == null || input_vectors == "") {
```
Can't we replace this part with "if (!input_vectors) "?

hy06ix commented on GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js:
```diff
@@ -40,25 +53,23 @@ module.exports = function (RED) {
-        typeof stored_vector == "undefined" ||
-        stored_vector == null ||
-        stored_vector == ""
+        typeof stored_vectors == "undefined" ||
```
Same question like line 60

hy06ix reviewed (dismissed):
Good change from single vector input to multiple vector input 👍

KimJaea committed:
fix: changed code to simpler version

KimJaea committed:
fix: add exception handling code

calculate-cosine.html:
if file path is empty, this node return false so it can alert user this node is incomplete. and also show alert message under the input box.

calculate-cosine.js:
if file path is not available, this node return text message. this message has Error topic, while normal result return message with Result topic.

KimJaea committed:
fix: exception handling without throwing error

if vector length is 0, return message topic error not throwing error

ramaking commented on GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js:
```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
+        if (!result) {
+          msg.payload = "The Vector is not available to calculate.";
+          msg.topic = "Error";
+          this.send(msg);
```
This code may cause unexpected type errors in the node responsible for output.
Don't use send method, just use node.error method and finish the node.

chickennight replied:
it will be better use 'node.error' in error case

KimJaea replied:
Checked! Now this handle every error by "this.error()". Please check new commit below 😄

Gitgloo commented on GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js:
```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
+
+        if (!result) {
+          msg.payload = "The Vector is not available to calculate.";
+          msg.topic = "Error";
```
why msg has 'topic' value?

KimJaea replied:
I used this to distinguish correct return value and others. But now I deleted it.

Gitgloo commented on GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js:
```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
+          this.send(msg);
+        } else {
+          msg.payload = result;
+          msg.topid = "Result";
```
I think 'topic' is right, not 'topid'

KimJaea committed:
fix: fixed handling error code

fixed handling error code. not using 'node.send', using 'node.error'.

KimJaea committed:
fix: delete topic from return msg
